woocommerce/woocommerce
# Trigger ajax add to cart JS event in case of response error

### All Submissions:






### Changes proposed in this Pull Request:


Add new event that's triggered on ajax add to cart response error. This new event would be useful for themes and plugins. Currently when the error redirection happens and then you click to back on page in the browser navigation, the 'added' and 'loading' states (class names) are not removed, which in some specific themes custom features turns the add to cart button disabled. Adding this new event would allow theme/plugins developers to do anything they want (like removing that 'added' and 'loading' classes) before the page redirection.



### How to test the changes in this Pull Request:




1. Install the Storefront theme.
2. Make sure ajax add to cart option is enabled.
3. Add new product and enable the option "Sold individually - Limit purchases to 1 item per order" option.
4. Go to the website Shop Catalog and add the product to cart twice. When you try to add the second time, you'll be redirected to the product page displaying an error message.
5. On Chrome or Safari, click in the browser back button.
6. The 'add to cart' button from the product will be unnacessible and with wrong 'loading' and 'added' classes, which makes the add to cart button unnacessible. (https://snipboard.io/gc2qT0.jpg)

The above instructions are about how to reproduce the issue with a default theme (Storefront). Here's a video showing the issue with another theme (Botiga) and the 'display quantity input on shop catalog' enabled: https://www.youtube.com/shorts/fJPXGMFe8tE



### Other information:




### FOR PR REVIEWER ONLY:
